nautobot/docs/user-guide/core-data-model/ipam/vlangroup.md

```diff
@@ -3,3 +3,177 @@
+
+## VLAN Group Ranges
+
++/- 2.4.0
+
+VLAN Groups contain a mandatory `range` field with a default value of `1-4094` which equals to all VLANs. This field may be used to constrain member VLANs of the group. Only VLANs with VLAN identifiers ("vids") within a `range` value will be allowed to associated with a `VLANGroup`
+
+Range value accepts commas and dashes, with examples as follows:
+
+* 1
+* 1-5
+* 1-5,10
+* 1-5,10-15
+* 1-5,10-15,16,17,18,19,20
+
+Values between dashes will also be expanded into a list of VLANs.
+
+## Creating new VLANs in a VLANGroup programatically
+
++/- 2.4.0
+
+VLAN Groups offer new API endpoint to list available and create new VLANs programatically. Nautobot accepts API calls to the following endpoint:
+
+`/api/ipam/vlan-groups/<vlan_group_id>/available-vlans/`
+
+For the `GET` request, API will return a list of all available VLAN IDs available in the given `VLANGroup`.
+
+For the `POST` request, API will create requested number of VLANs in a `VLANGroup`.
+
+### Examples
+
+To create a new `VLANGroup` with a `range` field of: `5-10` (permit VLANs `5, 6, 7, 8, 9, 10`), issue following POST request:
+
+```no-highlight
+curl -X POST -H "Authorization: Token $TOKEN" \
+-H "Content-Type: application/json" \
+-H "Accept: application/json; version=2.0; indent=4" \
+http://nautobot/api/ipam/vlan-groups/ \
+--data '[{"name": "Enterprise VLANs", "range": "5-10"}]'
+```
+
+`VLANGroup` was created as follows:
+
+```json
+[
+    {
+        "id": "c16d5d1d-40d6-4902-b89d-6294e60cb4bf",
+        "object_type": "ipam.vlangroup",
+        "display": "Enterprise VLANs",
+        "url": "http://nautobot/api/ipam/vlan-groups/c16d5d1d-40d6-4902-b89d-6294e60cb4bf/",
+        "natural_slug": "enterprise-vlans_c16d",
+        "name": "Enterprise VLANs",
+        "description": "",
+        "range": "5-10",
+        "location": null,
+        "created": "2024-09-25T09:27:16.343655Z",
+        "last_updated": "2024-09-25T09:27:16.343670Z",
+        "notes_url": "http://nautobot/api/ipam/vlan-groups/c16d5d1d-40d6-4902-b89d-6294e60cb4bf/notes/",
+        "custom_fields": {},
+        "computed_fields": {},
+        "relationships": {}
+    }
+]
+```
+
+Then, to list all available `VLANs` within an `Enterprise VLANs` `VLANGroup` issue a GET request:
+
+```no-highlight
+curl -X GET -H "Authorization: Token $TOKEN" \
+-H "Content-Type: application/json" \
+-H "Accept: application/json; version=2.0; indent=4" \
+http://nautobot/api/ipam/vlan-groups/c16d5d1d-40d6-4902-b89d-6294e60cb4bf/available-vlans/
+```
+
+```json
+{
+    "count": 6,
+    "next": null,
+    "previous": null,
+    "results": [
+        5,
+        6,
+        7,
+        8,
+        9,
+        10
+    ]
+
+```
+
+To create two new VLANs, a data payload has to be specified. In the case of first VLAN, its `VLAN ID` will be automatically determined based on `VLANGroups` availability, in the case of second VLAN we manually define `vid` requsting `VLAN` with `id` 8:
```

Comment:
Can we elaborate more on the data payload, i.e. mandatory `name` and `status`, optional `vid`, `tenant`, `role`, etc.?